crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -157,15 +160,34 @@ fn make_executor(kind: ExecutorKind) -> Box<dyn SystemExecutor> {
 }
 
 /// Chain systems into dependencies
-#[derive(PartialEq)]
-pub enum Chain {
-    /// Run nodes in order. If there are deferred parameters in preceeding systems a
-    /// [`apply_deferred`] will be added on the edge.
-    Yes,
-    /// Run nodes in order. This will not add [`apply_deferred`] between nodes.
-    YesIgnoreDeferred,
-    /// Nodes are allowed to run in any order.
-    No,
+#[derive(Default)]
+pub struct Chain(
+    /// Stores a map from the [`TypeId`] of the [`ScheduleBuildPass`] to the
+    /// corresponding boxed [`ScheduleBuildPass::EdgeOptions`], or None if
+    /// the systems are not chained.
+    Option<BTreeMap<TypeId, Box<dyn Any>>>,
```

Comment:
Question: why is this a `Box<dyn Any>` rather than `Box<dyn ScheduleBuildPassObj>`?

Reply:
Because it stores `ScheduleBuildPass::EdgeOptions` instead of `ScheduleBuildPassObj`